### PR TITLE
refactor(x/data): update sign to attest

### PR DESCRIPTION
**Pull request by ryanchristo**
## Description

Closes: #637

Updates the data module to use attestation terminology.

- Changes `SignData` to `Attest`
- Changes `Signer` to `Attestor`

Also changes `AnchorData` to `Anchor`.

---

### Author Checklist

*All items are required. Please add a note to the item if the item is not applicable and
please add links to any relevant follow up issues.*

I have...

- [ ] included the correct [type prefix](https://github.com/commitizen/conventional-commit-types/blob/v3.0.0/index.json) in the PR title
- [ ] added `!` to the type prefix if API or client breaking change
- [ ] targeted the correct branch (see [PR Targeting](https://github.com/cosmos/cosmos-sdk/blob/master/CONTRIBUTING.md#pr-targeting))
- [ ] provided a link to the relevant issue or specification
- [ ] followed the guidelines for [building modules](https://github.com/cosmos/cosmos-sdk/blob/master/docs/building-modules)
- [ ] included the necessary unit and integration [tests](https://github.com/cosmos/cosmos-sdk/blob/master/CONTRIBUTING.md#testing)
- [ ] added a changelog entry to `CHANGELOG.md`
- [ ] included comments for [documenting Go code](https://blog.golang.org/godoc)
- [ ] updated the relevant documentation or specification
- [ ] reviewed "Files changed" and left comments if necessary
- [ ] confirmed all CI checks have passed

### Reviewers Checklist

*All items are required. Please add a note if the item is not applicable and please add
your handle next to the items reviewed if you only reviewed selected items.*

I have...

- [ ] confirmed the correct [type prefix](https://github.com/commitizen/conventional-commit-types/blob/v3.0.0/index.json) in the PR title
- [ ] confirmed `!` in the type prefix if API or client breaking change
- [ ] confirmed all author checklist items have been addressed 
- [ ] reviewed state machine logic
- [ ] reviewed API design and naming
- [ ] reviewed documentation is accurate
- [ ] reviewed tests and test coverage
- [ ] manually tested (if applicable)

### CODE REVIEW

File: proto/regen/data/v1/tx.proto [hunk 1]
```diff
@@ -10,18 +10,18 @@ option go_package = "github.com/regen-network/regen-ledger/x/data";
 
 // Msg is the regen.data.v1alpha1 Msg service
 service Msg {
-  // AnchorData "anchors" a piece of data to the blockchain based on its secure
+  // Anchor "anchors" a piece of data to the blockchain based on its secure
```
Comment (robert-zaremba) [inline proto/regen/data/v1/tx.proto:13]:
```suggestion
  // Anchor a piece of data to the blockchain based on its secure
```

File: proto/regen/data/v1/tx.proto [hunk 2]
```diff
@@ -34,11 +34,11 @@ service Msg {
...
   //   and timestamping
   //
-  // SignData implicitly calls AnchorData if the data was not already anchored.
+  // Attest implicitly calls Anchor if the data was not already anchored.
   //
-  // SignData can be called multiple times for the same content hash with different
-  // signers and those signers will be appended to the list of signers.
-  rpc SignData(MsgSignData) returns (MsgSignDataResponse);
+  // Attest can be called multiple times for the same content hash with different
+  // attestors and those attestors will be appended to the list of attestors.
```
Comment (robert-zaremba) [inline proto/regen/data/v1/tx.proto:40]: What happens if the same attestor attests same data multiple times? Is it overwritten?

**Review by robert-zaremba — APPROVED** [review]
lgtm.
Let's update the `rpc Attest` doc.

File: proto/regen/data/v1/tx.proto (hunk 1 above)
Comment (ryanchristo) [reply proto/regen/data/v1/tx.proto:13]: `Anchor` is the name of the method here, not a verb.

File: proto/regen/data/v1/tx.proto (hunk 2 above)
Comment (ryanchristo) [reply proto/regen/data/v1/tx.proto:40]: In the current implementation, we don't do anything if an attestor attempts to attest to the same data. The original attestation with the original timestamp is not overwritten and no new attestation is added. I can add a note here.

**Pull request merged**